coding/coding_tests/trie_test.cpp

```diff
@@ -126,31 +152,54 @@ class CharValueList
   string m_string;
 };
 
-class Uint32ValueList
+class ValueListUint32
 {
 public:
-  using TBuffer = vector<uint32_t>;
+  using TValue = uint32_t;
+  using TSerializer = SingleValueSerializerUint32;
+
+  ValueListUint32() = default;
+
+  void Init(vector<TValue> const & values) { m_values = values; }
+
+  size_t Size() const { return m_values.size(); }
 
-  void Append(uint32_t value)
+  bool IsEmpty() const { return m_values.empty(); }
+
+  template <typename TSink>
+  void Serialize(TSink & sink, TSerializer const & /* serializer */) const
   {
-    m_values.push_back(value);
+    for (auto const & value : m_values)
+      WriteToSink(sink, value);
   }
 
-  uint32_t size() const { return m_values.size(); }
+  template <typename TSource>
+  void Deserialize(TSource & src, uint32_t valueCount, TSerializer const & /* serializer */)
```

Comment:
serializer can Serialize as well as Deserialize.
The problem looks common enough (cf. SerDes or https://msdn.microsoft.com/ru-ru/library/system.xml.serialization.xmlserializer(v=vs.110).aspx).
Do you have a better name?